sdk/core/core/src/az_credentials.c

```diff
@@ -62,8 +65,9 @@ AZ_NODISCARD az_result az_client_secret_credential_init(
     az_span client_secret) {
   *self = (az_client_secret_credential){
     ._internal = {
-      .vtbl = {
+      .credential = {
         ._internal = {
+          .http_client = NULL,
```

Comment:
this will be a transport_policy_options struct (not a pointer to it). http_client will be its only field (for now).